client/src/view/adjust-panel.tsx

```diff
@@ -194,6 +167,59 @@ export function AdjustPanel({fabric, setDisplacementSelection}: {
     )
 }
 
+function DisplacementSlider({adjustBars, displacementFromNuance, setFabricSlackLimits, nuance, setNuance}: {
+    adjustBars: boolean,
+    nuance: number,
+    setNuance: (nuance: number) => void,
+    displacementFromNuance: (nuance: number) => number,
+    setFabricSlackLimits: (bars: boolean, nuanceValue: number) => void,
+}): JSX.Element {
+
+    const [displacement, setDisplacement] = useState(0)
+
+    useEffect(() => {
+        const ticker = setInterval(() => setDisplacement(displacementFromNuance(nuance)), 1000)
+        return () => clearInterval(ticker)
+    }, [nuance])
+
+    const sliderStyle: CSSProperties = {
+        position: "relative",
+        height: "90%",
+        marginTop: "30%",
+        marginBottom: "30%",
+        borderRadius: 12,
+    }
+
+    const changeNuanceTo = (nuanceValue: number) => {
+        setNuance(nuanceValue)
+        setDisplacement(displacementFromNuance(nuanceValue))
+        setFabricSlackLimits(adjustBars, nuanceValue)
+    }
+
+    const onChange = (changedPercent: number[]) => {
+        const percent = changedPercent[0]
+        changeNuanceTo(percent / 100)
+    }
+
+    return (
+        <Slider rootStyle={sliderStyle} reversed={true} vertical={true}
+                step={-0.01} domain={DOMAIN} values={VALUES}
+                onChange={onChange}>
+            <Rail>{({getRailProps}) => <SliderRail getRailProps={getRailProps}/>}</Rail>
```

Comment:
Whoa how does this work? A function inside of a JSX element?

Reply:
It was the easiest slider for me to integrate.